src/components/header/EmptyThreadIndicator.js

```diff
@@ -0,0 +1,128 @@
+    emptyBufferStart: SyntheticCssDeclarations | null,
+  |} {
+    const {
+      rangeStart,
+      rangeEnd,
+      width,
+      unfilteredSamplesRange,
+      interval,
+      thread: { processShutdownTime, registerTime, unregisterTime },
+    } = this.props;
+    const rangeLength = rangeEnd - rangeStart;
+    const xPixelsPerMs = width / rangeLength;
+
+    const threadEndTime =
+      unregisterTime === null ? processShutdownTime : unregisterTime;
+
+    // Did this thread startup in this time range?
+    let startup = null;
+    if (registerTime > rangeStart) {
+      startup = {
+        left: 0,
+        width: (registerTime - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    // Did this thread shut down in this time range?
+    let shutdown = null;
+    if (threadEndTime !== null && threadEndTime < rangeEnd) {
+      shutdown = {
+        right: 0,
+        width: (rangeEnd - threadEndTime) * xPixelsPerMs,
+      };
+    }
+
+    let emptyBufferStart = null;
+    if (
+      // Threads could have no samples, and therefore no range.
+      unfilteredSamplesRange !== null &&
+      // Was the buffer empty at the beginning of the range, at least one interval length
+      // into the profile? This interval length ensures no awkward cut off where it's not
+      // really needed.
+      unfilteredSamplesRange.start > rangeStart + interval
+    ) {
+      const startMilliseconds = Math.max(0, registerTime - rangeStart);
+      emptyBufferStart = {
+        left: startMilliseconds * xPixelsPerMs,
+        right:
+          width - (unfilteredSamplesRange.start - rangeStart) * xPixelsPerMs,
+      };
+    }
+
+    return { startup, shutdown, emptyBufferStart };
+  }
+
+  render() {
+    const style = this._getIndicatorPositions();
+    return (
+      <div className="headerEmptyThreadIndicator">
+        {style.startup
+          ? <DivWithTooltip
+              style={style.startup}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorStartup"
+              tooltip="The thread was not yet started"
+            />
+          : null}
+        {style.shutdown
+          ? <DivWithTooltip
+              style={style.shutdown}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorShutdown"
+              tooltip="The thread was shut down here"
+            />
+          : null}
+        {style.emptyBufferStart
+          ? <DivWithTooltip
+              style={style.emptyBufferStart}
+              className="headerEmptyThreadIndicatorBlock headerEmptyThreadIndicatorEmptyBuffer"
+              tooltip={
+                <span className="headerEmptyThreadIndicatorLongTooltip">
+                  {oneLine`
+                    The buffer is empty. Either the profiler was still initializing for
+                    a new thread, or the profiling buffer was full. Increase your buffer
+                    size before recording to potentially include more samples.
```

Comment:
This is rather long, but I don't have a good suggestion.

Also I'm not sure about the sequence of tenses "is empty" vs "was full". You know better than me here so I'll defer to you.